app/src/main/java/com/mobileproto/hireddit/hireddit/visuals/SpeechFragment.java

```diff
@@ -77,64 +72,54 @@ public void onCreate(Bundle savedInstanceState)
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
-                             Bundle savedInstanceState)
-    {
+                             Bundle savedInstanceState) {
 
-        View view = inflater.inflate(R.layout.fragment_speak, container, false);
+        View view = inflater.inflate(R.layout.fragment_speech, container, false);
         ButterKnife.bind(this, view);
-        listener = new SpeechListener(new SpeechCallback()
-        {
+        listener = new SpeechListener(new SpeechCallback() {
             @Override
-            public void callback(ArrayList voiceResult)
-            {
+            public void callback(ArrayList voiceResult) {
                 voiceInput = voiceResult;
                 speechTextDisplay.setText(voiceInput.get(0).toString());
                 isListening = false;
+                Log.d(DEBUG_TAG, "Got result, stopped listening.");
             }
 
             @Override
-            public void errorCallback(int errorCode)
-            {
-                if (errorCode == SpeechRecognizer.ERROR_NO_MATCH) {
+            public void errorCallback(int errorCode, int numErrors) {
+                isListening = false;
+                Log.d(DEBUG_TAG, "Got error, stopped listening.");
+                if (errorCode == SpeechRecognizer.ERROR_NO_MATCH && numErrors == 1) { // error 7
+                    //numErrors is to check to make sure this is first time you get the error so you
+                    //don't get a bunch of toasts from unrecognized partial results
```

Comment:
They are notified - only for the first error though. this is because if theres a problem with recognizing what they said, youre going to get an error for each partial result and there'll be a ton of toasts.
numError is to just show the first error you get (youre either going to get 'no match' + 'general error'*x, 'say something' + 'general error'*x, or 'general error'*x [x meaning you get general error a bunch] at most, so it doesnt make sense to show more than just the first one.).

arraylist - as in, waiting until you're done with speech? onError is called every time theres an error